nilearn/image/image.py

```diff
@@ -880,6 +880,15 @@ def clean_img(imgs, sessions=None, detrend=True, standardize=True,
     from .image import new_img_like
 
     imgs_ = check_niimg_4d(imgs)
+
+    # Check if t_r is set, otherwise propse t_r from imgs header
+    if low_pass is not None or high_pass is not None:
+        if t_r is None:
+            t_r = imgs.header.get_zooms()[3]
+            raise ValueError(
```

Comment:
To be sure; Should we raise an error here or used the returned t_r ? Are we assuming that this t_r value is not reliable ?
(I think that BIDS conventions indeed require to provide the t_r value in an associated json file).

Reply:
I implemented this according to Gaël's comment [here](https://github.com/nilearn/nilearn/issues/1021#issuecomment-190209612), but I think the discussion is valid.
In my opinion, the `t_r` value is not reliable. For example, SPM12's slice timing correction resets the TR in the header to 0. The same can happen if users create new niftis, without setting the header correctly. Unfortunately, users rarely check that kind of things.
Therefore, a warning (only if temporal filters are applied), can point to bugs in a users code. Also, the BIDS convention wouldn't help with the automatic setting of `t_r`, but with the corresponding warning, users can verify that everything is correct.

Reply:
Makes sense. Maybe you want to add a short comment in the code stating this logic: namely that this value is considered as non-reliable, so that we chose to raise an error anyway.

Reply:
Good idea! Just did this with https://github.com/nilearn/nilearn/pull/1830/commits/ffa2167d681e8632a06d74c5f5af1678f15235ce.